### PR TITLE
Frrist/tests/migrate to v2 client

### LINKED ISSUE #4146: model.Execution and models.Execution have different definitions of terminal state
The model Execution considers the following states to be terminal:
- `ExecutionStateAskForBidRejected`
- `ExecutionStateBidRejected`
- `ExecutionStateCancelled`
- `ExecutionStateFailed`
- `ExecutionStateCompleted`

https://github.com/bacalhau-project/bacalhau/blob/17dbd8dac2fc47485b2a752f40d0d414078e6fe3/pkg/model/execution_state.go#L77-L79
 https://github.com/bacalhau-project/bacalhau/blob/17dbd8dac2fc47485b2a752f40d0d414078e6fe3/pkg/model/execution_state.go#L60-L63

The model Execution considers the following states to be terminal:
- `ExecutionStateBidRejected`
- `ExecutionStateCompleted`
- `ExecutionStateFailed`
- `ExecutionStateCancelled`

https://github.com/bacalhau-project/bacalhau/blob/17dbd8dac2fc47485b2a752f40d0d414078e6fe3/pkg/models/execution.go#L45-L50

The different being `ExecutionStateAskForBidRejected`. @wdbaruni is this a bug?

### PR DESCRIPTION
**Pull request by frrist**
Closes #4146, #4141, https://github.com/bacalhau-project/bacalhau/issues/4119, and #4143

### CODE REVIEW

File: pkg/models/execution.go [hunk 1]
```diff
@@ -46,7 +46,8 @@ func (s ExecutionStateType) IsTermainl() bool {
 	return s == ExecutionStateBidRejected ||
 		s == ExecutionStateCompleted ||
 		s == ExecutionStateFailed ||
-		s == ExecutionStateCancelled
+		s == ExecutionStateCancelled ||
+		s == ExecutionStateAskForBidRejected
 }
```
Comment (frrist) [inline pkg/models/execution.go:51]: fixes https://github.com/bacalhau-project/bacalhau/issues/4146

File: pkg/publicapi/endpoint/shared/endpoint.go [hunk 2]
```diff
@@ -78,6 +77,14 @@ func (e *Endpoint) nodeInfo(c echo.Context) error {
...
 }
 
+type VersionRequest struct {
+	ClientID string `json:"client_id" example:"ac13188e93c97a9c2e7cf8e86c7313156a73436036f30da1ececc2ce79f9ea51"`
+}
+
+type VersionResponse struct {
+	VersionInfo *models.BuildVersionInfo `json:"build_version_info"`
+}
+
```
Comment (wdbaruni) [inline pkg/publicapi/endpoint/shared/endpoint.go:87]: lets move all api models to `publicapi/apimodels`

Comment (wdbaruni) [reply pkg/publicapi/endpoint/shared/endpoint.go:87]: actually we already have `apimodels.GetVersionResponse` and I think we can deprecate most of the APIs under `endpoint.shared`, specifically `node_info` and `version`

Comment (wdbaruni) [reply pkg/publicapi/endpoint/shared/endpoint.go:87]: and lets not use `ClientID` if we can

**Review by wdbaruni — APPROVED** [review]
LGTM. Just minor comments

File: pkg/publicapi/endpoint/shared/endpoint.go (hunk 2 above)
Comment (wdbaruni) [reply pkg/publicapi/endpoint/shared/endpoint.go:87]: actually it seems we need `/` to return the node version for better user experience. Lets maintain that behaviour, but avoid code duplication

File: pkg/publicapi/endpoint/shared/endpoint.go [hunk 3]
```diff
@@ -78,6 +77,14 @@ func (e *Endpoint) nodeInfo(c echo.Context) error {
...
 // version godoc
```
Comment (udsamani) [inline pkg/publicapi/endpoint/shared/endpoint.go:88]: The godoc needs to change. 

`@Param VersionRequest body VersionRequest`

**Pull request merged**